File: src/main/java/me/zax71/lightHub/inventories/gameMenu.java
package me.zax71.lightHub.inventories;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.TextComponent;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextDecoration;
import net.minestom.server.entity.Player;
import net.minestom.server.inventory.Inventory;
import net.minestom.server.inventory.InventoryType;
import net.minestom.server.inventory.TransactionOption;
import net.minestom.server.item.ItemStack;
import net.minestom.server.item.Material;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class gameMenu {



    public static void open(Player player){

        TextComponent inventoryTitle = Component.text("Music Discs", NamedTextColor.WHITE);
        Inventory inventory = new Inventory(InventoryType.CHEST_3_ROW, inventoryTitle);

        // Create list for items
        List<ItemStack> itemStacks = new

        ItemStack GRAY_GLASS = ItemStack.builder(Material.GRAY_STAINED_GLASS_PANE).displayName(Component.text("").decoration(TextDecoration.ITALIC, false)).build();


        // Add dark gray glass to list
        for (int i = 0; i < inventory.getSize(); i++) {
            itemStacks.add(GRAY_GLASS);
        }

        // inventory.addItemStacks(itemStacks, TransactionOption.ALL);

        // Add list to inventory with each item representing one slot
        for (int i = 0; i < itemStacks.size(); i++) {
            inventory.setItemStack(i, itemStacks.get(i));
        }

        inventory.copyContents(itemStacks);

        player.openInventory(inventory);
    }


}

File: src/main/kotlin/me/zax71/lightHub/inventories/GameMenu.kt
package me.zax71.lightHub.inventories

import me.zax71.lightHub.config
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextDecoration
import net.kyori.adventure.text.minimessage.MiniMessage
import net.minestom.server.inventory.Inventory
import net.minestom.server.inventory.InventoryType
import net.minestom.server.item.ItemStack
import net.minestom.server.item.Material
import net.minestom.server.tag.Tag
import org.slf4j.Logger

object GameMenu {

    var inventory: Inventory? = null

    init {
        val inventoryTitle = Component.text("Game Menu", NamedTextColor.GRAY)
        inventory = Inventory(InventoryType.CHEST_3_ROW, inventoryTitle)

        val grayGlass = ItemStack.builder(Material.GRAY_STAINED_GLASS_PANE)
            .displayName(Component.text("").decoration(TextDecoration.ITALIC, false)).build()

        // Create list for items
        val itemStacks = Array(inventory!!.size) { grayGlass }

        itemStacks[22] = ItemStack.builder(Material.BARRIER)
            .displayName(
                Component.text("Close", NamedTextColor.RED).decoration(TextDecoration.ITALIC, false)
            )
            .build()

        // Get config values and put them in to lists
        for (i in 0..8) {
            val typeConfig = config.getString("gameMenu.items.$i.type")
            val serverConfig = config.getString("gameMenu.items.$i.server")
            val nameConfig = config.getString("gameMenu.items.$i.name")

            // Stay as glass when null
            if (typeConfig == null || typeConfig == "") {
                continue
            }

            // String -> Material
            val typeMaterial = Material.fromNamespaceId(typeConfig.lowercase())
            if (typeMaterial == null) {
                org.tinylog.kotlin.Logger.error {"gameMenu.items.$i.name is $typeConfig and that is not a valid Material"}
                break
            }

            val item = ItemStack.builder(typeMaterial)
                .displayName(MiniMessage.miniMessage().deserialize(nameConfig))
                .set(Tag.String("server"), serverConfig)
                .build()

            itemStacks[i+9] = item

        }

        // Add items to inv
        inventory!!.copyContents(itemStacks)

        inventory!!.addInventoryCondition { player, slot, _, inventoryConditionResult ->
            // Moving items is bad
            inventoryConditionResult.isCancel = true


            if (slot == 22) {
                player.closeInventory()
            }
        }
    }
}